compiler/src/main/scala/org/scalajs/core/compiler/JSEncoding.scala

```diff
@@ -148,12 +148,15 @@ trait JSEncoding extends SubComponent { self: GenJSCode =>
 
     def name = encodeMemberNameInternal(sym)
 
+    def privateSuffix: String =
+      if (sym.owner.isTraitOrInterface) encodeClassFullName(sym.owner)
```

Comment:
We'll get plenty of tests like that for free with 2.12.0-M4. I did hit bugs, and that's how I found out I had to change this.

We cannot test this before 2.12.0-M4 because nothing can ever generate a *private* default methods.